CMake Error: 'arm64' is an unsupported architecture

Hi friends,

I'm trying to open dotnet runtime coreclr directory with CLion, but I got the below error, how to fix this.

```
/Applications/CLion.app/Contents/bin/cmake/mac/aarch64/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_MAKE_PROGRAM=/Applications/CLion.app/Contents/bin/ninja/mac/aarch64/ninja -G Ninja -S /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/src/coreclr -B /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/src/coreclr/cmake-build-debug
-- The C compiler identification is AppleClang 16.0.0.16000026
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/functions.cmake:7 (message):
  'arm64' is an unsupported architecture.
Call Stack (most recent call first):
  /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configureplatform.cmake:112 (clr_unknown_arch)
  /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configuretools.cmake:1 (include)
  /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configurecompiler.cmake:5 (include)
  CMakeLists.txt:7 (include)


-- Configuring incomplete, errors occurred!

[Finished]

```

--- a/src/coreclr/CMakePresets.json
+++ b/src/coreclr/CMakePresets.json
@@ -40,8 +40,7 @@
       "vendor": {
         "microsoft.com/VisualStudioSettings/CMake/1.0": {
           "hostOS": [
-            "macOS",
-            "OSX"
+            "macOS"
           ]
         }
       }

--- a/src/coreclr/CMakePresets.json
+++ b/src/coreclr/CMakePresets.json
@@ -38,10 +38,7 @@
         "CLR_CMAKE_TARGET_OS": "darwin"
       },
       "vendor": {
-        "microsoft.com/VisualStudioSettings/CMake/1.0": {
-          "hostOS": [
-            "macOS"
-          ]
+        "jetbrains.com/clion": {
         }
       }
     },

--- a/src/coreclr/CMakePresets.json
+++ b/src/coreclr/CMakePresets.json
@@ -215,7 +215,7 @@
     },
     {
       "name": "osx.arm64.Release",
-      "displayName": "macos.arm64.Release",
+      "displayName": "osx.arm64.Release",
       "inherits": [
         "osx-base",
         "Release",

--- a/src/coreclr/CMakePresets.json
+++ b/src/coreclr/CMakePresets.json
@@ -31,6 +31,21 @@
       }
     },
     {
+      "name": "osx-base",
+      "hidden": true,
+      "inherits": "base",
+      "cacheVariables": {
+        "CLR_CMAKE_TARGET_OS": "darwin"
+      },
+      "vendor": {
+        "microsoft.com/VisualStudioSettings/CMake/1.0": {
+          "hostOS": [
+            "macOS"
+          ]
+        }
+      }
+    },
+    {
       "name": "Debug",
       "hidden": true,
       "cacheVariables": {
@@ -161,6 +176,60 @@
       "displayName": "windows.arm64.Checked",
       "inherits": [
         "windows-base",
+        "Checked",
+        "ARM64"
+      ]
+    },
+    {
+      "name": "osx.x64.Debug",
+      "displayName": "osx.x64.Debug",
+      "inherits": [
+        "osx-base",
+        "Debug",
+        "x64"
+      ]
+    },
+    {
+      "name": "osx.x64.Release",
+      "displayName": "osx.x64.Release",
+      "inherits": [
+        "osx-base",
+        "Release",
+        "x64"
+      ]
+    },
+    {
+      "name": "osx.x64.Checked",
+      "displayName": "osx.x64.Checked",
+      "inherits": [
+        "osx-base",
+        "Checked",
+        "x64"
+      ]
+    },
+    {
+      "name": "osx.arm64.Debug",
+      "displayName": "osx.arm64.Debug",
+      "inherits": [
+        "osx-base",
+        "Debug",
+        "ARM64"
+      ]
+    },
+    {
+      "name": "osx.arm64.Release",
+      "displayName": "macos.arm64.Release",
+      "inherits": [
+        "osx-base",
+        "Release",
+        "ARM64"
+      ]
+    },
+    {
+      "name": "osx.arm64.Checked",
+      "displayName": "osx.arm64.Checked",
+      "inherits": [
+        "osx-base",
         "Checked",
         "ARM64"
       ]

--- a/src/coreclr/CMakePresets.json
+++ b/src/coreclr/CMakePresets.json
@@ -38,7 +38,11 @@
         "CLR_CMAKE_TARGET_OS": "darwin"
       },
       "vendor": {
-        "jetbrains.com/clion": {
+        "microsoft.com/VisualStudioSettings/CMake/1.0": {
+          "hostOS": [
+            "macOS",
+            "OSX"
+          ]
         }
       }
     },

